### PR TITLE
Spec SSV tests alignment

### PR DESCRIPTION
**Pull request by AKorpusenko**

### CODE REVIEW

File: eth/ethtest/utils_test.go [hunk 1]
```diff
@@ -10,8 +10,8 @@ import (
 	ethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/golang/mock/gomock"
 	"github.com/herumi/bls-eth-go-binary/bls"
-	gomock "go.uber.org/mock/gomock"
```
Comment (MatheusFranco99) [inline eth/ethtest/utils_test.go:14]: Same here @MatusKysel?

File: eth/eventhandler/event_handler_test.go [hunk 2]
```diff
@@ -18,10 +18,10 @@ import (
 	ethcommon "github.com/ethereum/go-ethereum/common"
 	ethtypes "github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/golang/mock/gomock"
 	"github.com/herumi/bls-eth-go-binary/bls"
 	"github.com/pkg/errors"
 	"github.com/stretchr/testify/require"
-	gomock "go.uber.org/mock/gomock"
```
Comment (MatheusFranco99) [inline eth/eventhandler/event_handler_test.go:24]: And here @MatusKysel?

File: protocol/v2/types/ssvshare.go [hunk 3]
```diff
@@ -96,6 +105,10 @@ func ComputeQuorumAndPartialQuorum(committeeSize int) (quorum uint64, partialQuo
 	return uint64(f*2 + 1), uint64(f + 1)
 }
 
+func ComputeF(committeeSize int) uint64 {
+	return uint64(committeeSize-1) / 3
+}
```
Comment (MatheusFranco99) [inline protocol/v2/types/ssvshare.go:110]: Since we're defining a function to compute f, I think we can re-use it in `ComputeQuorumAndPartialQuorum` and `ValidCommitteeSize` to avoid knowledge duplication

File: protocol/v2/ssv/validator/committee.go [hunk 4]
```diff
@@ -36,7 +37,8 @@ type Committee struct {
 	//sharesMtx sync.RWMutex
 	Shares map[phase0.ValidatorIndex]*spectypes.Share
 
-	Operator                *spectypes.Operator
```
Comment (MatheusFranco99) [inline protocol/v2/ssv/validator/committee.go:39]: In spec, we also renamed the attribute `Operator` to `CommitteeMember`. But this was last week so perhaps this change is not in the version that was used as reference for this PR

File: protocol/v2/ssv/validator/opts.go [hunk 5]
```diff
@@ -23,7 +23,7 @@ type Options struct {
 	BeaconNetwork     beacon.BeaconNetwork
 	Storage           *storage.QBFTStores
 	SSVShare          *types.SSVShare
-	Operator          *spectypes.Operator
```
Comment (MatheusFranco99) [inline protocol/v2/ssv/validator/opts.go:26]: I would suggest the same rename here (`Operator` to `CommitteeMember`). If there's already the intention to do this in another PR, it's fine :)

File: protocol/v2/ssv/spectest/sync_committee_aggregator_proof_type.go [hunk 6]
```diff
@@ -73,13 +73,13 @@ func overrideStateComparisonForSyncCommitteeAggregatorProofSpecTest(t *testing.T
...
 func keySetForShare(share *spectypes.Share) *testingutils.TestKeySet {
-	if share.Quorum == 5 {
+	if len(share.Committee) == 5 {
 		return testingutils.Testing7SharesSet()
 	}
-	if share.Quorum == 7 {
+	if len(share.Committee) == 7 {
 		return testingutils.Testing10SharesSet()
 	}
-	if share.Quorum == 9 {
```
Comment (MatheusFranco99) [inline protocol/v2/ssv/spectest/sync_committee_aggregator_proof_type.go:82]: Here the length of the Committee is compared to 5, 7 and 9. But these are actually quorum sizes. In spec, we have:
```go
func KeySetForShare(share *types.Share) *TestKeySet {
	if len(share.Committee) == 7 {
		return Testing7SharesSet()
	}
	if len(share.Committee) == 10 {
		return Testing10SharesSet()
	}
	if len(share.Committee) == 13 {
		return Testing13SharesSet()
	}
	return Testing4SharesSet()
}
```

File: protocol/v2/ssv/spectest/multi_start_new_runner_duty_type.go [hunk 7]
```diff
@@ -97,6 +100,27 @@ func (test *StartNewRunnerDutySpecTest) RunAsPartOfMultiTest(t *testing.T, logge
...
+					panic(err) // should never occur given previous filter to ensure valid JSON
+				}
+				return out
+			}),
+			),
+		)
+	if len(diff) > 0 {
+		logJSON(t, "test_SHOULDBE_PostDutyRunnerState_"+test.Name, test.PostDutyRunnerState)
+		logJSON(t, "test_HAVE_Runner"+test.Name, test.Runner)
+	}
```
Comment (MatheusFranco99) [inline protocol/v2/ssv/spectest/multi_start_new_runner_duty_type.go:123]: I think that this was used for debugging. Do you want this to be kept?

File: protocol/v2/ssv/spectest/committee_msg_processing_type.go [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+	// post root
+	postRoot, err := test.Committee.GetRoot()
+	require.NoError(t, err)
+
+	if test.PostDutyCommitteeRoot != hex.EncodeToString(postRoot[:]) {
+		diff := cmp.Diff(test.Committee, test.PostDutyCommittee, cmp.Exporter(func(p reflect.Type) bool { return true }))
+		//// DEBUG
+		//logJSON(t, "test_committee#"+test.Name+"#", test.Committee)
+		//logJSON(t, "test_post_duty_committee#"+test.Name+"#", test.PostDutyCommittee)
+		//
```
Comment (MatheusFranco99) [inline protocol/v2/ssv/spectest/committee_msg_processing_type.go:81]: I believe this may be removed

File: protocol/v2/ssv/spectest/committee_msg_processing_type.go [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+	}
+}
+
+// TODO: REMOVE when tests alignment is finished
+func logJSON(t *testing.T, name string, value interface{}) {
+	bytes, err := json.Marshal(value)
+	require.NoError(t, err)
+	err = os.WriteFile(fmt.Sprintf("%s_test_serialized.json", name), bytes, 0644)
+	require.NoError(t, err)
+}
```
Comment (MatheusFranco99) [inline protocol/v2/ssv/spectest/committee_msg_processing_type.go:92]: Also here (regarding removing)

File: protocol/v2/qbft/controller/controller.go [hunk 10]
```diff
@@ -26,22 +26,74 @@ type Controller struct {
...
 }
 
+//func (b *Controller) MarshalJSON() ([]byte, error) {
+//	type ControllerAlias struct {
+//		State              *State
+//		Share              map[phase0.ValidatorIndex]*spectypes.Share
+//		QBFTController     *controller.Controller
+//		BeaconNetwork      spectypes.BeaconNetwork
+//		RunnerRoleType     spectypes.RunnerRole
+//		highestDecidedSlot phase0.Slot
```
Comment (MatheusFranco99) [inline protocol/v2/qbft/controller/controller.go:42]: Perhaps this commented code can be removed, no?

**Comment by AKorpusenko** [comment]
replaced this PR with: https://github.com/ssvlabs/ssv/pull/1453
